### DISCUSSION TITLE
Selecting rows such that there exists a relationship such that ...

### DISCUSSION THREAD

**Discussion by diracdeltafunk** [supabase/supabase#16009] [Questions]
I have 3 tables:
* `objects`
* `properties`
* `relationships`

`objects` contains a list of objects I want to keep track of, `properties` contains a list of properties these objects may or may not have, and `relationships` keeps track of whether or not each object has each property. More precisely, `relationships` has four columns:
* a primary key (serial)
* a column `object` with a foreign key relation to the primary key of `objects`
* a column `property` with a foreign key relation to the primary key of `properties`
* a boolean column `value`

Thus, each row of `relationships` specifies a pair of an object and a property, and whether or not the specified object has the specified property.

---

Now, let's say I want to select those objects which satisfy property 1 and property 2 but not property 3.

Using the js library, I can at least do this:

```
supabase
    .from('objects')
    .select('*, relationships(id:object)', { count: 'exact' })
```

which returns all the rows of `objects`, along with all relationships which reference that object. However, I don't know how to filter from here. How do I specify that I only want the rows of `objects` such that

(at least one of these relationships has `property=1` and `value=TRUE`)
and
(at least one of these relationships has `property=2` and `value=TRUE`)
and
(at least one of these relationships has `property=3` and `value=FALSE`)

?

---

Thanks in advance for your help :)

**Comment by raulrpearson** [comment]
Taking a stab at this, probably doesn't work as-is, but maybe it points you in the right direction:

```javascript
supabase
    .from('objects')
    .select('*, relationships(id:object, property, value)', { count: 'exact' })
    .or('property.eq.1, value.eq.true', { foreignTable: 'properties' })
    .or('property.eq.2, value.eq.true', { foreignTable: 'properties' })
    .or('property.eq.3, value.eq.false', { foreignTable: 'properties' })
```

Check out the [or](https://supabase.com/docs/reference/javascript/or) section from the JS docs, it has an "or with and" and a "or on foreign tables" examples that are were I got the idea from.
